MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs

```diff
@@ -25,76 +25,81 @@ namespace MediaBrowser.XbmcMetadata.Savers
-                    "imdbid",
-                    "imdb_id",
-                    "country",
-                    "audiodbalbumid",
-                    "audiodbartistid",
-                    "enddate",
-                    "lockedfields",
-                    "zap2itid",
-                    "tvrageid",
-
-                    "musicbrainzartistid",
-                    "musicbrainzalbumartistid",
-                    "musicbrainzalbumid",
-                    "musicbrainzreleasegroupid",
-                    "tvdbid",
-                    "collectionitem",
-
-                    "isuserfavorite",
-                    "userrating",
-
-                    "countrycode"
-
-        }.ToDictionary(i => i, StringComparer.OrdinalIgnoreCase);
+        public const string DateAddedFormat = "yyyy-MM-dd HH:mm:ss";
+
+        public const string YouTubeWatchUrl = "https://www.youtube.com/watch?v=";
+
+        private static readonly CultureInfo _usCulture = CultureInfo.ReadOnly(new CultureInfo("en-US"));
+
+        private static readonly HashSet<string> _commonTags = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "plot",
+            "customrating",
+            "lockdata",
+            "dateadded",
+            "title",
+            "rating",
+            "year",
+            "sorttitle",
+            "mpaa",
+            "aspectratio",
+            "collectionnumber",
+            "tmdbid",
+            "rottentomatoesid",
+            "language",
+            "tvcomid",
+            "tagline",
+            "studio",
+            "genre",
+            "tag",
+            "runtime",
+            "actor",
+            "criticrating",
+            "fileinfo",
+            "director",
+            "writer",
+            "trailer",
+            "premiered",
+            "releasedate",
+            "outline",
+            "id",
+            "credits",
+            "originaltitle",
+            "watched",
+            "playcount",
+            "lastplayed",
+            "art",
+            "resume",
+            "biography",
+            "formed",
+            "review",
+            "style",
+            "imdbid",
+            "imdb_id",
+            "country",
+            "audiodbalbumid",
+            "audiodbartistid",
+            "enddate",
+            "lockedfields",
+            "zap2itid",
+            "tvrageid",
+
+            "musicbrainzartistid",
+            "musicbrainzalbumartistid",
+            "musicbrainzalbumid",
+            "musicbrainzreleasegroupid",
+            "tvdbid",
+            "collectionitem",
+
+            "isuserfavorite",
+            "userrating",
+
+            "countrycode"
+        };
+
+        // filters control characters but allows only properly-formed surrogate sequences
+        private const string _invalidXMLCharsRegex = @"(?<![\uD800-\uDBFF])[\uDC00-\uDFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x9F\uFEFF\uFFFE\uFFFF]";
+
```

Comment:
double newline again, can we add a rule to warn for this?

Reply:
I think there is one, it's just a bit hard to spot with 13000 warnings.